tests/release/test_fixture_args.py

```diff
@@ -0,0 +1,96 @@
+from pathlib import Path
+
+import pytest
+
+from pseudopeople.interface import (
+    generate_american_community_survey,
+    generate_current_population_survey,
+    generate_decennial_census,
+    generate_social_security,
+    generate_taxes_1040,
+    generate_taxes_w2_and_1099,
+    generate_women_infants_and_children,
+)
+from tests.release.conftest import (
+    DEFAULT_ENGINE,
+    DEFAULT_STATE,
+    DEFAULT_YEAR,
+    FULL_USA_FILEPATH,
+    RI_FILEPATH,
+    _parse_dataset_params,
+)
+
+
+@pytest.fixture()
+def check_subprocess_environment() -> None:
+    if "RUNNING_AS_SUBPROCESS" not in os.environ:
+        pytest.skip("Skipping this test because it's not running as a subprocess")
+
+
+EXPECTED_PARAMETERS = {
+    "census": (
+        generate_decennial_census,
+        None,
+        DEFAULT_ENGINE,
+        DEFAULT_STATE,
+        DEFAULT_YEAR,
+    ),
+    "acs": (
+        generate_american_community_survey,
+        FULL_USA_FILEPATH,
+        DEFAULT_ENGINE,
+        DEFAULT_STATE,
+        DEFAULT_YEAR,
+    ),
+    "ssa": (generate_social_security, None, "dask", DEFAULT_STATE, DEFAULT_YEAR),
+    "cps": (generate_current_population_survey, None, DEFAULT_ENGINE, "RI", DEFAULT_YEAR),
+    "tax_w2_1099": (
+        generate_taxes_w2_and_1099,
+        None,
+        DEFAULT_ENGINE,
+        DEFAULT_STATE,
+        2010,
+    ),
+    "wic": (generate_women_infants_and_children, None, DEFAULT_ENGINE, "RI", 2010),
+    "tax_1040": (
+        generate_taxes_1040,
+        RI_FILEPATH,
+        DEFAULT_ENGINE,
+        DEFAULT_STATE,
+        DEFAULT_YEAR,
+    ),
+}
+
+
+@pytest.mark.subprocess_test
+@pytest.mark.usefixtures("check_subprocess_environment")
+def test_parsing_fixture_params(request: pytest.FixtureRequest) -> None:
+    output = _parse_dataset_params(request)
+    # we know output will have a string as the first element but can't type this
+    # while specifying the types of the other elements in output
+    dataset_name: str = output[0]  # type: ignore [assignment]
+    assert output[1:] == EXPECTED_PARAMETERS[dataset_name]
+
+
+@pytest.mark.parametrize(
+    "pytest_args",
+    [
+        (["--dataset", "census"]),
+        (["--dataset", "acs", "--population", "USA"]),
+        (["--dataset", "tax_1040", "--population", "RI"]),
+        (["--dataset", "ssa", "--engine", "dask"]),
+        (["--dataset", "cps", "--state", "RI"]),
+        (["--dataset", "tax_w2_1099", "--year", "2010"]),
+        (["--dataset", "wic", "--state", "RI", "--year", "2010"]),
+    ],
+)
+def test_parsing_fixture_param_combinations(pytest_args: list[str]) -> None:
+    env = os.environ.copy()
+    env["RUNNING_AS_SUBPROCESS"] = "1"
+    os.chdir(Path(__file__).parent)  # need this to access options from conftest.py
+    base_cmd = ["pytest", "-k", "test_parsing_fixture_params"]
+    cmd = base_cmd + pytest_args
+    result = subprocess.run(cmd, capture_output=True, text=True, env=env)
+    assert result.returncode == 0
```

Comment:
I think you want to unset the `RUNNING_AS_SUBPROCESS` env var at the end of this test. pytest runs top to bottom so this should work as is, but if we ever move this test _above_ the `test_parsing_fixture_params` then that would try and run a second time and presumably fail